### PR TITLE
Pre Submission

### PR DESCRIPTION
**Pull request by acowap**

### CODE REVIEW

File: HW-3.Rmd [hunk 1]
```diff
@@ -56,7 +57,7 @@ correctly predicted.
 # To keep this exercise simple, let's remove the only 3 users (0.005% of users)
 # who did not list a height, define the outcome variable, and add an ID variable
 # to distinguish the users
-profiles <- read_csv(file="profiles.csv") %>% 
+profiles <- read_csv(file="CopyOfprofiles.csv") %>%
```
Comment (rudeboybert) [inline HW-3.Rmd]: teachable moment in reproducibility. I don't have access to `CopyOfprofiles.csv`, thus my file wouldn't Knit. To resolve this, you could've included this file in your commit, but given how large this file is, the easiest thing to do is just set this to `profiles.csv`. Keep this in my for the final project when sharing code and data.

File: HW-3.Rmd [hunk 2]
```diff
@@ -80,7 +81,22 @@ to rate how well we performed.
...
-set.seed(76)
+
+profiles <- profiles %>% mutate(diet3 = ifelse(grepl("anything", diet), "anything", diet)) %>% 
+  mutate(diet3 = ifelse(grepl("vegetarian", diet), "vegetarian", diet3)) %>% 
+  mutate(diet3 = ifelse(grepl("vegan", diet), "vegan", diet3)) %>% 
+  mutate(diet3 = ifelse(grepl("kosher", diet), "other", diet3)) %>% 
+  mutate(diet3 = ifelse(grepl("halal", diet), "other", diet3)) %>% 
+  mutate(diet3 = ifelse(grepl("other", diet), "other", diet3)) %>% 
+  mutate(diet3 = ifelse(is.na(diet), "na", diet3)) %>% 
+  mutate(job2 = ifelse(is.na(job), "na", job)) %>%
```
Comment (rudeboybert) [inline HW-3.Rmd]: ok, but keep in mind, your predictor variable is essentially "listed job vs not" i.e. for the people who listed their job, you are losing information on *which* job they listed.

File: HW-3.Rmd [hunk 3]
```diff
@@ -80,7 +81,22 @@ to rate how well we performed.
...
+  mutate(diet3 = ifelse(grepl("other", diet), "other", diet3)) %>% 
+  mutate(diet3 = ifelse(is.na(diet), "na", diet3)) %>% 
+  mutate(job2 = ifelse(is.na(job), "na", job)) %>% 
+  mutate(income2 = ifelse(is.na(income), "na", income))
+  
+
+
+training <- profiles[sample(nrow(profiles), 2997), ]
+
+test <- profiles[sample(nrow(profiles), 56946), ]
```
Comment (rudeboybert) [inline HW-3.Rmd]: your training and test set will most likely have overlap here! You want to partition them into 2 distinct data sets. Take a look at line 63 above for a hint as to how. What's the opposite of a `_join`?

File: HW-3.Rmd [hunk 4]
```diff
@@ -107,20 +126,31 @@ for each users' sex, then rate how well the model performs on the `test` data.
 type="response")` does? The help file is located in `?predict.glm`
 
 ```{r, echo=TRUE, message=TRUE, warning=TRUE, fig.width=8, fig.height=4.5}
+pre_train <- predict(predict_sex_model)
 
-```
+train_pre <- training %>% mutate(phat1 = pre_train) %>% mutate(prediction = ifelse(phat1 >.5, 1, 0)) %>%
```
Comment (rudeboybert) [inline HW-3.Rmd]: coding aesthetic nitpick: please have a unique piped operation per line. It makes it much easier for me to read:

```
predictions <- test %>% 
  mutate(phat1 = pre) %>% 
  mutate(prediction = ifelse(phat1 >.5, 1, 0)) %>% 
  mutate(correct = ifelse(prediction == is_female, 1, 0))
```

File: HW-3.Rmd [hunk 5]
```diff
@@ -135,11 +165,49 @@ Let the reference currency be US dollars. Analyze these results and provide
...
+ggplot(data = bitcoin_vol, aes(x =Date, y=volatility_b)) + geom_point() + 
+  labs(title = "Bitcoin Volatility")
+
+
+avg_g_vol <- tail(gold$Value, -1) - head(gold$Value, -1)
+avg_g_vol <- append(avg_g_vol, 0, after = 0)
+
+gold_vol <- gold %>% mutate(volatility_g = avg_g_vol)
+
+ggplot(data = gold_vol, aes(x = Date, y = volatility_g)) + geom_point() + labs(title = "Gold Volatility")
```
Comment (rudeboybert) [inline HW-3.Rmd]: while technically not wrong, [time series](https://en.wikipedia.org/wiki/Time_series) are almost universally done with `geom_line()`, since there is an obvious dependency of an observation at time `t` with both the observation at time `t-1` and `t+1`

File: HW-3.Rmd [hunk 6]
```diff
@@ -135,11 +165,49 @@ Let the reference currency be US dollars. Analyze these results and provide
...
+Avg_vol <- append(Avg_vol, 0, after=0)
 
+bitcoin_vol <- bitcoin %>% mutate(volatility_b = Avg_vol)
+
+ggplot(data = bitcoin_vol, aes(x =Date, y=volatility_b)) + geom_point() + 
+  labs(title = "Bitcoin Volatility")
+
+
+avg_g_vol <- tail(gold$Value, -1) - head(gold$Value, -1)
+avg_g_vol <- append(avg_g_vol, 0, after = 0)
```
Comment (rudeboybert) [inline HW-3.Rmd]: A bit hacky. Consider a more elegant solution using the `lead()` and `lag()` function on the back of dplyr cheatsheet, middle column, window functions

File: HW-3.Rmd [hunk 7]
```diff
@@ -135,11 +165,49 @@ Let the reference currency be US dollars. Analyze these results and provide
...
+
+ggplot(data = gold_vol, aes(x = Date, y = volatility_g)) + geom_point() + labs(title = "Gold Volatility")
+
+interval <- interval(ymd("2010-01-01"), ymd("2016-10-21"))
+
+vol <- full_join(gold_vol, bitcoin_vol, by = "Date") %>% filter(Date %within% interval)
+
+plot <- ggplot(data = vol, aes(x = Date)) + labs(title = "both", y = "Volatility")
+
+plot + geom_line(aes(y=volatility_g, color = "gold")) + geom_line(aes(y = volatility_b, color = "bitcoin"))
```
Comment (rudeboybert) [inline HW-3.Rmd]: A gain of 41.94% from 2015-01-12 to 2015-01-13 in `Avg`? Are you sure?

Related to my confusion, this is where having the data in tidy format, instead of working with two separate data frames, really pays off. A single data frame with columns `Date`, `Volatility`, and `Type`

Speak to me in person if you don't know what I'm talking about.

File: HW-3.Rmd [hunk 8]
```diff
@@ -135,11 +165,49 @@ Let the reference currency be US dollars. Analyze these results and provide
...
+
+
+```
+
+
+While both Gold and Bitcoin are generally stable, Bitcoin started off more stable than gold from 2010-2013, but more recently Bitcoin has become more volatile than gold. Bitcoin has also faced much larger volatility shocks (as in 2014). Therefore, gold is a more stable investment than Bitcoin.
```
Comment (rudeboybert) [inline HW-3.Rmd]: You need to describe what your measure is!

File: HW-3.Rmd [hunk 9]
```diff
@@ -178,6 +246,14 @@ the "graveyard shift" during the academic year? Define
 
 ```{r, echo=TRUE, message=TRUE, warning=TRUE, fig.width=8, fig.height=4.5}
 
+jukebox2 <- jukebox %>% mutate(dt = parse_date_time(date_time, "a b d HMS y")) %>% 
+  mutate(month = month(dt)) %>% mutate(hour = hour(dt)) %>% 
+  filter(month <=5 | month >=9) %>% filter(hour==24 | hour < 8) %>% 
+  group_by(artist) %>% tally() %>% arrange(desc(n)) %>% head(10)
+  
+jukebox2
```
Comment (rudeboybert) [inline HW-3.Rmd]: on the surface, this doesn't look wrong.

**Comment by rudeboybert** [comment]
looks good so far.

File: HW-3.Rmd [hunk 10]
```diff
@@ -135,11 +165,38 @@ Let the reference currency be US dollars. Analyze these results and provide
...
-```
+gold <- Quandl("BUNDESBANK/BBK01_WT5511") %>%
+  tbl_df() 
+
+bitcoin_vol <- bitcoin %>% mutate(Avglag = lag(Avg)) %>% mutate(volatility_b = (Avg-Avglag)/Avg)
+
+ggplot(data = bitcoin_vol, aes(x =Date, y=volatility_b)) + geom_line() + 
+  labs(title = "Bitcoin Volatility", y="Volatility")
 
+gold_vol <- gold %>% mutate(Valuelag = lag(Value)) %>% mutate(volatility_g = (Value-Valuelag)/Value)
```
Comment (rudeboybert) [inline HW-3.Rmd]: consider multiplying by 100 to make it % change.

Comment (rudeboybert) [inline HW-3.Rmd]: always do a quick sanity check to make sure that the ascending/decending dates and lead/lag aren't giving you the opposite sign of true change.

Comment (rudeboybert) [inline HW-3.Rmd]: while not essential for this HW, consider keeping things in tidy format (date, type, vol) so that you're assured all packages play nice.

File: HW-3.Rmd [hunk 11]
```diff
@@ -80,7 +79,20 @@ to rate how well we performed.
...
+  mutate(diet3 = ifelse(grepl("vegetarian", diet), "vegetarian", diet3)) %>% 
+  mutate(diet3 = ifelse(grepl("vegan", diet), "vegan", diet3)) %>% 
+  mutate(diet3 = ifelse(grepl("kosher", diet), "other", diet3)) %>% 
+  mutate(diet3 = ifelse(grepl("halal", diet), "other", diet3)) %>% 
+  mutate(diet3 = ifelse(grepl("other", diet), "other", diet3)) %>% 
+  mutate(diet3 = ifelse(is.na(diet), "na", diet3)) %>% 
+  mutate(job2 = ifelse(is.na(job), "na", job)) %>% 
+  mutate(income2 = ifelse(is.na(income), "na", income))
+  
+training <- profiles[sample(nrow(profiles), 2997), ]
```
Comment (rudeboybert) [inline HW-3.Rmd]: consider using `dplyr::sample_n` for computational efficiency.

File: HW-3.Rmd [hunk 12]
```diff
@@ -80,7 +79,20 @@ to rate how well we performed.
...
+
+test <- profiles %>% filter(!(id %in% training$id))
```
Comment (rudeboybert) [inline HW-3.Rmd]: again, anti_join(by="id").

File: HW-3.Rmd [hunk 13]
```diff
@@ -107,20 +122,35 @@ for each users' sex, then rate how well the model performs on the `test` data.
 type="response")` does? The help file is located in `?predict.glm`
 
 ```{r, echo=TRUE, message=TRUE, warning=TRUE, fig.width=8, fig.height=4.5}
+pre_train <- predict(predict_sex_model)
```
Comment (rudeboybert) [inline HW-3.Rmd]: `pre_train <- predict(predict_sex_model, newdata = training, type = "response")`
because we need to apply inverse logit function.

File: HW-3.Rmd [hunk 14]
```diff
@@ -107,20 +122,35 @@ for each users' sex, then rate how well the model performs on the `test` data.
...
 
-#### d)
+predictions <- test %>% 
+  mutate(phat1 = pre) %>% 
+  mutate(prediction = ifelse(phat1 >.5, 1, 0)) %>% 
+  mutate(correct = ifelse(prediction == is_female, 1, 0))
 
-Did the model perform better on the `training` data or the `test` data? Why
-do you think that is?
+p_correct_test <- (sum(predictions$correct)/56946)*100
```
Comment (rudeboybert) [inline HW-3.Rmd]: `nrow(test)` in 56946

File: HW-3.Rmd [hunk 15]
```diff
@@ -107,20 +122,35 @@ for each users' sex, then rate how well the model performs on the `test` data.
...
+  mutate(prediction = ifelse(phat1 >.5, 1, 0)) %>% 
+  mutate(correct = ifelse(prediction == is_female, 1, 0))
 
 
+pre <- predict(predict_sex_model, newdata = test, type = "response")
 
-#### d)
+predictions <- test %>% 
+  mutate(phat1 = pre) %>% 
+  mutate(prediction = ifelse(phat1 >.5, 1, 0)) %>%
```
Comment (rudeboybert) [inline HW-3.Rmd]: the 0.5 is called a decision threshold.
